include/fiction/utils/round_n_decimal_places.hpp

```diff
@@ -0,0 +1,31 @@
+//
+// Created by Jan Drewniok on 19.04.23.
+//
+
+#ifndef FICTION_ROUND_N_DECIMAL_PLACES_HPP
+#define FICTION_ROUND_N_DECIMAL_PLACES_HPP
+
+#include <cmath>
+
+namespace fiction
+{
+
+/**
+ * Rounds a number to a specified number of decimal places.
+ *
+ * @tparam T the type of the number to round.
+ * @param number the number to round.
+ * @param n the number of decimal places to round to.
+ * @return the number rounded to n decimal places.
+ */
+
+template <typename T>
+T round_n_decimal_places(T number, const uint64_t n)
+{
```

Comment:
We can add a `static_assert` for `std::is_arithmetic<T>` to ensure that `T` is a number type.